bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,460 @@
+			} else {
+				processedData[ key ] = flattenData( value );
+			}
+		}
+
+		return processedData;
+	}
+
+	function flattenData( nestedData ) {
+		let flatData = [];
+
+		for ( const value of Object.values( nestedData ) ) {
+			if ( Array.isArray( value ) ) {
+				flatData = [ ...flatData, ...value ];
+			} else {
+				flatData = [ ...flatData, ...flattenData( value ) ];
+			}
+		}
+
+		return flatData;
+	}
+
+	return processLevel( data );
+}
+
+function processCommits( commits ) {
+	const result = {};
+
+	commits.forEach( ( commit ) => {
+		// Skip commits without an associated pull request
+		if ( ! commit.pullRequest ) {
+			return;
+		}
+		commit.files.forEach( ( file ) => {
+			// Skip files that are not PHP files.
+			if ( ! file.filename.endsWith( '.php' ) ) {
+				return;
+			}
+
+			// Skip files within specific packages.
+			if (
+				IGNORED_PATHS.some(
+					( path ) =>
+						file.filename.startsWith( path ) ||
+						file.filename === path
+				)
+			) {
+				return;
+			}
+
+			const parts = file.filename.split( '/' );
+			let current = result;
+
+			// If the file is under 'phpunit', add it directly to the 'phpunit' key
+			// this is it's helpful to have a full list of commits that modify tests.
+			if ( parts.includes( 'phpunit' ) ) {
+				current.phpunit = current.phpunit || [];
+				current.phpunit = [ ...current.phpunit, commit ];
+				return;
+			}
+
+			for ( let i = 0; i < parts.length; i++ ) {
+				const part = parts[ i ];
+
+				// Skip 'src' part under 'block-library'
+				if ( part === 'src' && parts[ i - 1 ] === 'block-library' ) {
+					continue;
+				}
+
+				if ( i === parts.length - 1 ) {
+					current[ part ] = current[ part ] || [];
+					current[ part ] = [ ...current[ part ], commit ];
+				} else {
+					current[ part ] = current[ part ] || {};
+					current = current[ part ];
+				}
+			}
+		} );
+	} );
+
+	return result;
+}
+
+function formatPRLine( pr ) {
+	return `- [ ] ${ pr.url } - @/${ pr.creator } | Trac ticket | Core backport PR\n`;
+}
+
+function formatHeading( level, key ) {
+	const emoji = key.endsWith( '.php' ) ? '📄' : '📁';
+	return `${ '#'.repeat( level ) } ${ emoji } ${ key }\n\n`;
+}
+
+function generateIssueContent( result, level = 1 ) {
+	let issueContent = '';
+	let isFirstSection = true;
+
+	for ( const [ key, value ] of Object.entries( result ) ) {
+		// Add horizontal rule divider between sections, but not before the first section
+		if ( level <= 2 && ! isFirstSection ) {
+			issueContent += '\n---\n';
+		}
+
+		issueContent += formatHeading( level, key );
+
+		if ( Array.isArray( value ) ) {
+			value.forEach( ( commit ) => {
+				issueContent += formatPRLine( commit.pullRequest );
+			} );
+		} else {
+			issueContent += generateIssueContent( value, level + 1 );
+		}
+
+		isFirstSection = false;
+	}
+
+	return issueContent;
+}
+
+async function getAllCommits( octokit, owner, repo, since, path ) {
+	let commits = [];
+	if ( DEBUG ) {
```

Comment:
We should probably remove this or use a dedicated flag for "low-network" or something/